packages/utils/ast-utils.test.js

```diff
@@ -210,4 +210,146 @@ const a = { plugs: [] }
 			expect(j(require).toSource()).toMatchSnapshot();
 		});
 	});
+
+	describe("safeTraverse", () => {
+		it("should safe traverse", () => {
+			const type = {
+				type: "NodeType"
+			};
+			const p = {
+				value: {
+					value: type
+				}
+			};
+			const require = utils.safeTraverse(p, ["value", "value"]);
+			expect(require).toEqual(type);
+		});
+
+		it("should safe traverse thrice", () => {
+			const type = {
+				type: "NodeType"
+			};
+			const p = {
+				parent: {
+					value: {
+						value: type
+					}
+				}
+			};
+			const require = utils.safeTraverse(p, ["parent", "value", "value"]);
+			expect(require).toEqual(type);
+		});
+	});
+
+	describe("safeTraverseAndGetType", () => {
+		it("should safe traverse and get type", () => {
+			const NODE_TYPE = "NodeType";
+			const p = {
+				value: {
+					value: {
+						type: NODE_TYPE
+					}
+				}
+			};
+			const require = utils.safeTraverseAndGetType(p);
+			expect(require).toEqual(NODE_TYPE);
+		});
+
+		it("should safe traverse and return false if not found", () => {
+			const NODE_TYPE = "NodeType";
+			const p = {
+				ed: {
+					sheeran: {
+						type: NODE_TYPE
+					}
+				}
+			};
+			const require = utils.safeTraverseAndGetType(p);
+			expect(require).toEqual(false);
```

Comment:
this is not a valid test, you're testing the node against itself. Give it a real path to check. `p`, `foo`, `bar`.